mlflow/store/artifact/databricks_logged_model_artifact_repo.py

```diff
@@ -0,0 +1,156 @@
+import logging
+import posixpath
+import re
+from concurrent.futures import ThreadPoolExecutor
+from pathlib import Path
+from typing import Optional
+
+from databricks.sdk import WorkspaceClient
+from databricks.sdk.config import Config
+from databricks.sdk.errors.platform import NotFound
+from databricks.sdk.service.files import FilesAPI
+
+from mlflow.entities import FileInfo
+from mlflow.exceptions import MlflowException
+from mlflow.store.artifact.artifact_repo import ArtifactRepository
+from mlflow.store.artifact.databricks_artifact_repo import DatabricksArtifactRepository
+
+_logger = logging.getLogger(__name__)
+
+
+class DatabricksLoggedModelArtifactRepository(ArtifactRepository):
+    # Matches URIs of the form:
+    # databricks/mlflow-tracking/<experiment_id>/logged_models/<model_id>/<relative_path>
+    _URI_REGEX = re.compile(
+        r"databricks/mlflow-tracking/(?P<experiment_id>[^/]+)/logged_models/(?P<model_id>[^/]+)(?P<relative_path>/.*)?$"
+    )
+
+    @staticmethod
```

Comment:
According to the repository's `.rules/misisng-types.md` rule, this method should include type hints. Please update to:

```python
@staticmethod
def is_logged_model_uri(artifact_uri: str) -> bool:
    return bool(DatabricksLoggedModelArtifactRepository._URI_REGEX.search(artifact_uri))
```